packages/core/src/storages/request_provider.ts

```diff
@@ -332,47 +347,45 @@ export abstract class RequestProvider implements IStorage {
             }),
         );
 
-        // The `requests` array can be huge, and `ow` is very slow for anything more complex.
-        // This explicit iteration takes a few milliseconds, while the ow check can take tens of seconds.
-
-        // ow(requests, ow.array.ofType(ow.any(
-        //     ow.string,
-        //     ow.object.partialShape({ url: ow.string, id: ow.undefined }),
-        //     ow.object.partialShape({ requestsFromUrl: ow.string, regex: ow.optional.regExp }),
-        // )));
-
-        for (const request of requests) {
-            if (typeof request === 'string') {
-                continue;
-            }
-
-            if (typeof request === 'object' && request !== null) {
-                if (typeof request.url === 'string' && typeof request.id === 'undefined') {
-                    continue;
-                }
-
-                if (typeof (request as any).requestsFromUrl === 'string') {
-                    continue;
+        const addRequest = this.addRequest.bind(this);
+
+        async function* generateRequests() {
+            for await (const opts of asyncifyIterable(requests)) {
+                // The `requests` array can be huge, and `ow` is very slow for anything more complex.
+                // This handwritten check takes a few milliseconds, while the ow check can take tens of seconds.
```

Comment:
This feels redundant nowadays, can just yeet it

Reply:
Do you mean the "validation" as a whole? Or just the comment?